### PR TITLE
MAINT: add spotless code formatting -- first pass

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#574 

*Description of changes:*

-----
This PR 

(1) Applied spotless code formatting to all file types
(2) Introduced the basic rules.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chenqi0805** [comment]
Get a blocking bug:
```
Cannot cast object 'extension 'docker'' with class 'com.bmuschko.gradle.docker.DockerExtension_Decorated' to class 'com.bmuschko.gradle.docker.DockerExtension'
```
It seems to stem from some conflict between spotless and existing com.bmuschko.docker-remote-api plugin.

**Comment by wrijeff** [comment]
> Get a blocking bug:
> 
> ```
> Cannot cast object 'extension 'docker'' with class 'com.bmuschko.gradle.docker.DockerExtension_Decorated' to class 'com.bmuschko.gradle.docker.DockerExtension'
> ```
> 
> It seems to stem from some conflict between spotless and existing com.bmuschko.docker-remote-api plugin.

Appears to have the same root cause as https://github.com/opendistro-for-elasticsearch/data-prepper/pull/551. Not sure how the upgraded version is getting pulled in on this PR though. Fixing the prior issue should unblock this one.

**Comment by chenqi0805** [comment]
Turns out by removing 
```
plugins {
    id 'com.bmuschko.docker-remote-api' version '6.6.1'
}
```
from the root gradle file `data-prepper-core/build.gradle` the issus is resolved.

File: build.gradle [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
     repositories {
         mavenCentral()
         maven { url "https://jitpack.io" }
     }
+
+    spotless {
+        format 'misc', {
+            target '*.md', '.gitignore', '*.yml', '*.yaml'
+            // TODO: enrich format rules
+            trimTrailingWhitespace()
```
Comment (wrijeff) [inline build.gradle:20]: We might want to _not_ trim whitespace on markdown files. From https://daringfireball.net/projects/markdown/syntax#block:

>"When you do want to insert a \<br /> break tag using Markdown, you end a line with two or more spaces, then type return."

Would just need to confirm if the formatter would strip double-spaces which were intended to be used as line breaks. Or we can just encourage people to not use double-spaces in our files. 😉 

I only see one double-space being removed in this PR, and I don't think it was intended to be a line break. Can probably merge as is and test locally.

**Review by wrijeff — APPROVED** [review]
👍

File: build.gradle (hunk 1 above)
Comment (chenqi0805) [reply build.gradle:20]: > We might want to not trim whitespace on markdown files. From https://daringfireball.net/projects/markdown/syntax#block:

Sure. Will reserve markdown in a separate block.

> Would just need to confirm if the formatter would strip double-spaces which were intended to be used as line breaks. Or we can just encourage people to not use double-spaces in our files.

No. Tested it does not.

**Pull request merged**